src/assets/scripts/client/info/AirportInfoController.js

```diff
@@ -0,0 +1,287 @@
+                ._setupHandlers()
+                ._enable();
+    }
+
+    /**
+     * @for AirportInfoController
+     * @method init
+     * @chainable
+     */
+    init() {
+        this.simClockController = new SimClockController();
+        this.$template = $(AIRPORT_INFO_TEMPLATE);
+        this.$clockView = this.$template.find(INFO_VIEW_SELECTORS.CLOCK_VALUE);
+        this.$windView = this.$template.find(INFO_VIEW_SELECTORS.WIND_VALUE);
+        this.$altimeterView = this.$template.find(INFO_VIEW_SELECTORS.ALTIMETER_VALUE);
+        this.$elevationView = this.$template.find(INFO_VIEW_SELECTORS.ELEVATION_VALUE);
+
+        return this;
+    }
+
+    /**
+     * @for AirportInfoController
+     * @method _setupHandlers
+     * @chainable
+     * @private
+     */
+    _setupHandlers() {
+        this.onAirportChangeHandler = this.onAirportChange.bind(this);
+
+        return this;
+    }
+
+    /**
+     * Set initial element references
+     *
+     * Should be run once only on instantiation
+     *
+     * @for StripViewModel
+     * @method _createChildren
+     * @chainable
+     * @private
+     */
+    _createChildren() {
+        this.$element.append(this.$template);
+
+        return this;
+    }
+
+    /**
+     * Enable all event handlers
+     *
+     * @for AirportInfoController
+     * @method _enable
+     * @chainable
+     * @private
+     */
+    _enable() {
+        this._eventBus.on(EVENT.AIRPORT_CHANGE, this.onAirportChangeHandler);
+        this.onAirportChange();
+
+        return this;
+    }
+
+    /**
+     * Disable all event handlers and destroy the instance
+     *
+     * @for AirportInfoController
+     * @method _disable
+     * @chainable
+     * @private
+     */
+    _disable() {
+        this.$element = null;
+        this.$template = null;
+        this.simClockController = null;
+        this.airportInfoController = null;
+
+        this._eventBus.off(EVENT.AIRPORT_CHANGE, this._onAirportChangeHandler);
+
+        return this;
+    }
+
+    /**
+     * @for AirportInfoController
+     * @method reset
+     * @chainable
+     */
+    reset() {
+        this.wind = null;
+        this.altimeter = null;
+        this.elevation = null;
+        this.icao = null;
+
+        return this;
+    }
+
+    /**
+     * Updates the information taken from the AirportModel: the wind, the altimeter,
+     * and the elevation. Triggered on airport change.
+     *
+     * @for AirportInfoController
+     * @method onAirportChange
+     */
+    onAirportChange() {
+        const airport = AirportController.airport_get();
+        const icao = airport.icao.toUpperCase();
+        const windAngle = Math.round(radiansToDegrees(airport.wind.angle));
+
+        this.wind = this._buildWindAndGustReadout({ speed: airport.wind.speed, angle: windAngle });
+        this.altimeter = this._generateAltimeterReading(airport.wind.speed);
+        this.elevation = `${airport.elevation}`;
+        this.icao = icao;
```

Comment:
this should be normalized to lowercase and only transformed to uppercase for the view.  that way you can avoid all the `toLowerCase` and/or `toUpperCase` calls.  There aren't many here, true, but doing it once on the way in (if at all since the `AirportModel` *should* already do this) makes things much more consistent.

We currently don't do this well and it's a big pain in the ass.  TL;DR strings should *always* be stored in lower case and only transformed to uppercase in the view.

Reply:
In this case, `icao` is only used to display the airport code.